### ISSUE TITLE
Save a txt file that shows train history for each run (matterport and yolo)

### ISSUE THREAD

**Issue by joshdejeu** [joshdejeu/CSC-537-Final#3]
Since matterport and yolo weight files get overwritten every epoch it would be helpful to hook into the training loop after each epoch and get metric data such.

Example with matterport after each epoch:
`56s 556ms/step - loss: 0.1347 - rpn_class_loss: 0.0024 - rpn_bbox_loss: 0.0191 - mrcnn_class_loss: 0.0125 - mrcnn_bbox_loss: 0.0133 - mrcnn_mask_loss: 0.0873 - val_loss: 0.8699 - val_rpn_class_loss: 0.0152 - val_rpn_bbox_loss: 0.3223 - val_mrcnn_class_loss: 0.1219 - val_mrcnn_bbox_loss: 0.1969 - val_mrcnn_mask_loss: 0.2136`

At minimum keeping track of every time a run is started/resume with what settings:
How many epochs?
Batch size?
Which layers are getting trained?

Possibly, log every new best weight to track improvements

This would help to visualize the evaluations for every model and keep record of the history for a specific run
